crates/subspace-networking/src/request_response_handler.rs

```diff
@@ -39,25 +42,63 @@ use sc_peerset::ReputationChange;
 use crate::request_responses::{IncomingRequest, OutgoingResponse, ProtocolConfig};
 const LOG_TARGET: &str = "request-response-handler";
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Request {
+    pub start: PieceIndexHash,
+}
```

Comment:
I found it confusing that so specific use case is exposed as generic `Request` data structure via public API.